File: src/FlashcardParser/SingleLineParser/Resource.ts
import { LineDescriptor, IStudySet } from "../Types/Types";

export function parseResourceLine(
    lineDescriptor: LineDescriptor,
    studySet: IStudySet
): boolean {
    const { trimmedLine } = lineDescriptor;

    studySet.resources[trimmedLine] = trimmedLine;
    if (!studySet.defaultResource) {
        console.log(`[parser] Using resource as default`);
        studySet.defaultResource = trimmedLine;
    }

    return true;
}

File: service/flashcard_parser/single_line_parser/resource.py
from ..types import LineDescriptor, StudySet


def parse_resource_line(line_descriptor: LineDescriptor, study_set: StudySet) -> bool:
    trimmed = line_descriptor.trimmed_line
    study_set.resources[trimmed] = trimmed
    if not study_set.default_resource:
        print("[parser] Using resource as default")
        study_set.default_resource = trimmed
    return True
